src/rdkafka_partition.c

```diff
@@ -2855,6 +2900,20 @@ rd_kafka_topic_partition_list_add(rd_kafka_topic_partition_list_t *rktparlist,
 }
 
 
+rd_kafka_topic_partition_t *rd_kafka_topic_partition_list_add_with_topic_id(
+    rd_kafka_topic_partition_list_t *rktparlist,
+    rd_kafka_Uuid_t topic_id,
+    int32_t partition) {
+        rd_kafka_topic_partition_t *rktpar;
+        rktpar = rd_kafka_topic_partition_list_add0(
+            __FUNCTION__, __LINE__, rktparlist, NULL, partition, NULL, NULL);
+        rd_kafka_topic_partition_private_t *parpriv =
+            rd_kafka_topic_partition_get_private(rktpar);
+        parpriv->topic_id = topic_id;
```

Comment:
Shall we check `RD_KAFKA_UUID_ZERO` case here like we are doing in `set` and `get` topic_id? First let's understand why do we need this check there in the first place.

Reply:
I can add an optimization if it's `RD_KAFKA_UUID_ZERO`